service/history/workflowResetor.go

```diff
@@ -530,7 +531,8 @@ func (w *workflowResetorImpl) replayReceivedSignals(
 		for {
 			var readResp *persistence.ReadHistoryBranchByBatchResponse
 			readResp, err := w.eng.historyV2Mgr.ReadHistoryBranchByBatch(readReq)
-			if err != nil {
+			// Fail if we don't have enough events to perform the reset, otherwise continue with what we've got.
+			if err != nil && (readResp == nil || readResp.LastEventID < workflowTaskFinishEventID) {
```

Comment:
trying to figure out when readResp would actually be nil here. It seems we always initialize the struct and return a pointer to it, right?

Reply:
Yes, just a safety check.